src/app/clusters/window-covering-server/window-covering-server.cpp

```diff
@@ -0,0 +1,651 @@
+    WindowCover & cover  = WindowCover::Instance();
+    bool hasLift         = cover.hasFeature(WindowCover::Feature::Lift);
+    bool isPositionAware = cover.hasFeature(WindowCover::Feature::PositionAware);
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+
+    emberAfWindowCoveringClusterPrint("GoToLiftValue Value command received");
+    if (hasLift && isPositionAware)
+    {
+        cover.Lift().TargetSet(value, WindowCover::PositionUnits::Value);
+    }
+    else
+    {
+        status = EMBER_ZCL_STATUS_ACTION_DENIED;
+        emberAfWindowCoveringClusterPrint("Err Device is not PA=%u or LF=%u", isPositionAware, hasLift);
+    }
+    emberAfSendImmediateDefaultResponse(status);
+    return true;
+}
+
+/**
+ * @brief Window Covering Cluster Go To Lift Percentage Command callback
+ * @param percentLiftValue
+ */
+
+bool emberAfWindowCoveringClusterGoToLiftPercentageCallback(chip::app::CommandHandler * cmmd, uint8_t percent)
+{
+    WindowCover & cover  = WindowCover::Instance();
+    bool hasLift         = cover.hasFeature(WindowCover::Feature::Lift);
+    bool isPositionAware = cover.hasFeature(WindowCover::Feature::PositionAware);
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+
+    emberAfWindowCoveringClusterPrint("GoToLiftPercentage Percentage command received");
+    if (hasLift && isPositionAware)
+    {
+        cover.Lift().TargetSet(percent, WindowCover::PositionUnits::Percentage);
+    }
+    else
+    {
+        status = EMBER_ZCL_STATUS_ACTION_DENIED;
+        emberAfWindowCoveringClusterPrint("Err Device is not PA=%u or LF=%u", isPositionAware, hasLift);
+    }
+    emberAfSendImmediateDefaultResponse(status);
+    return true;
+}
+
+bool __attribute__((weak))
+emberAfWindowCoveringClusterGoToLiftPercentageCallback(chip::app::CommandHandler * cmmd, uint8_t percent, uint16_t percent100ths)
+{
+    WindowCover & cover  = WindowCover::Instance();
+    bool hasLift         = cover.hasFeature(WindowCover::Feature::Lift);
+    bool isPositionAware = cover.hasFeature(WindowCover::Feature::PositionAware);
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+
+    emberAfWindowCoveringClusterPrint("GoToLiftPercentage Percentage command received");
+    if (hasLift && isPositionAware)
+    {
+        if (percent > 100)
+        {
+            cover.Lift().TargetSet(percent100ths, WindowCover::PositionUnits::Percentage100ths);
+        }
+        else
+        {
+            cover.Lift().TargetSet(percent, WindowCover::PositionUnits::Percentage);
+        }
+    }
+    else
+    {
+        status = EMBER_ZCL_STATUS_ACTION_DENIED;
+        emberAfWindowCoveringClusterPrint("Err Device is not PA=%u or LF=%u", isPositionAware, hasLift);
+    }
+    emberAfSendImmediateDefaultResponse(status);
+    return true;
+}
+
+/**
+ * @brief Window Covering Cluster Go To Tilt Value Command callback
+ * @param tiltValue
+ */
+
+bool emberAfWindowCoveringClusterGoToTiltValueCallback(chip::app::CommandHandler * cmmd, uint16_t value)
+{
```

Comment:
I think there might be a threading problem because this callbacks are done through the Interaction Modle Engine and the app will also use the same instance. And this are two separate threads.

Reply:
this is noted and we will take care of it later